votor/src/consensus_metrics.rs

```diff
@@ -2,16 +2,37 @@
 
 use {
     agave_votor_messages::vote::Vote,
+    crossbeam_channel::{Receiver, RecvTimeoutError, Sender},
     histogram::Histogram,
     solana_clock::{Epoch, Slot},
     solana_metrics::datapoint_info,
     solana_pubkey::Pubkey,
     std::{
         collections::BTreeMap,
+        sync::{
+            atomic::{AtomicBool, Ordering},
+            Arc,
+        },
+        thread::Builder,
         time::{Duration, Instant},
     },
 };
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ConsensusMetricsEvent {
+    /// A vote was received from the node with `id`.
+    Vote { id: Pubkey, vote: Vote },
+    /// A block hash was seen for `slot` and the `leader` is responsible for producing it.
+    BlockHashSeen { leader: Pubkey, slot: Slot },
+    /// Check on new epoch
+    MaybeNewEpoch { epoch: Epoch },
+    /// Start of slot
+    StartOfSlot { slot: Slot },
+}
+
+pub type ConsensusMetricsEventSender = Sender<(Instant, Vec<ConsensusMetricsEvent>)>;
+pub type ConsensusMetricsEventReceiver = Receiver<(Instant, Vec<ConsensusMetricsEvent>)>;
```

Comment:
nit: this type alias is not saving a lot of characters of typing.  And more often than not, I have to look up the alias definition when reading code.  I would prefer not having them.

Reply:
Maybe let's wait a little bit. Because later we need to pass the senders around inside TVU, the code there would be easier to read and easier to change using type alias.